### PR TITLE
WARN: sort_values for midx warns unnecessarily if nan is in level

### PR DESCRIPTION
**Pull request by phofl**
I think we can skip the whatsnew here, this caused only a RuntimeWarning before

### CODE REVIEW

File: pandas/tests/indexes/multi/test_sorting.py [hunk 1]
```diff
@@ -338,3 +338,12 @@ def test_sort_values_with_na_na_position(dtype, na_position):
         ]
     expected = MultiIndex.from_arrays(arrays)
     tm.assert_index_equal(result, expected)
+
+
+def test_sort_unnecessary_warning():
+    # GH#55386
+    midx = MultiIndex.from_tuples([(1.5, 2), (3.5, 3), (0, 1)])
+    midx = midx.set_levels([2.5, np.nan, 1], level=0)
+    result = midx.sort_values()
```
Comment (mroeschke) [inline pandas/tests/indexes/multi/test_sorting.py:347]: Should we wrap this in a `tm.assert_produces_warning(False)`?

Comment (phofl) [reply pandas/tests/indexes/multi/test_sorting.py:347]: Ci would raise anyway, that’s how I found this

Comment (mroeschke) [reply pandas/tests/indexes/multi/test_sorting.py:347]: Ah okay, sounds good

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**